pkg/controller/helper/finalizers.go

```diff
@@ -0,0 +1,36 @@
+package helper
+
+import (
+	"context"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+/*
+This function reconciles the finalizers, it requires
+- object
+- k8client
+- finalizer
+If the deletion timestamp is found, the finalizer will be removed.
```

Comment:
If we remove the finalizer on finding the deletion timestamp, will it allow for the finalization to happen? I think we want to carry out some finalizing logic before removing it. Were you